core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java

```diff
@@ -661,4 +643,98 @@ public static String getTitleBlockName(String type) {
     }
     return "";
   }
+
+  /**
+   * Returns the evaluation result of a Title Block expression (aql expression or capella semantic browser query).
+   * 
+   * @param descriptor
+   * @param expression
+   * @param titleBlock
+   * @return the evaluation result of a Title Block expression.
+   */
+  public static Object getResultOfExpression(DRepresentationDescriptor descriptor, String expression,
+      DAnnotation titleBlock) {
+    EObject semanticElement = TitleBlockHelper.getSemanticElementReference(titleBlock);
+
+    // if is a Diagram Title Block, the semantic element will be the descriptor
+    if (semanticElement == null) {
+      semanticElement = descriptor;
+    }
+
+    IInterpreterProvider provider = CompoundInterpreter.INSTANCE.getProviderForExpression(expression);
+
+    if (provider instanceof DefaultInterpreterProvider) {
+      return new EvaluationException();
+    }
+
+    IInterpreter interpreter = provider.createInterpreter();
+    Object result = null;
+    try {
+      result = interpreter.evaluate(semanticElement, expression);
+    } catch (EvaluationException e) {
+      return e;
+    }
+
+    if (result instanceof Collection) {
+      Collection<?> originalResult = (Collection<?>) result;
+      return sanitizeResultItems(originalResult);
+    }
+
+    return sanitizeResultItem(result);
+  }
+
+  private static boolean isValidResultItem(Object item) {
+    if (item instanceof EObject) {
+      EObject eObject = (EObject) item;
+
+      if (eObject instanceof Element || eObject instanceof DRepresentationDescriptor) {
+        return true;
+      }
+
+      EPackage ePackage = eObject.eClass().getEPackage();
```

Comment:
Could you extract this code into a separate method (e.g. isViewpointElement)?

Reply:
Done